airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py

```diff
@@ -154,17 +158,22 @@ def _calculate_cursor_datetime_from_state(self, stream_state: Mapping[str, Any])
             return self.parse_date(stream_state[self.cursor_field.eval(self.config)])
         return datetime.datetime.min.replace(tzinfo=datetime.timezone.utc)
 
-    def _format_datetime(self, dt: datetime.datetime):
+    def _format_datetime(self, dt: datetime.datetime) -> str:
         return self._parser.format(dt, self.datetime_format)
 
+    def _format_slice_datetime(self, dt: datetime.datetime) -> str:
+        if not self.cursor_datetime_formats:
+            raise RuntimeError("DatetimeBasedCursor can't operate without cursor_datetime_formats")
```

Comment:
When would this happen? When we instatiate the DatetimeBasedCursor, we set it to `self.cursor_datetime_formats = [self.datetime_format]` if cursor_datetime_formats is empty. And since `datetime_format` is a required field it'll always have a value

Reply:
It should not be an issue. I removed this code